src/java.base/share/classes/java/io/ByteArrayInputStream.java

```diff
@@ -207,10 +207,21 @@ public int readNBytes(byte[] b, int off, int len) {
     public synchronized long transferTo(OutputStream out) throws IOException {
         int len = count - pos;
         if (len > 0) {
+            byte[] tmp;
+            if (out.getClass().getPackageName().startsWith("java.") &&
```

Comment:
Has anybody actually estimated or measured if such an exception is actually useful / needed given the fact that System.arraycopy is fast native code and most buffers used by java.io-located streams are just few KB? Just asking as it could be the case that *interpreting* this Java bytecode could be slower than executing some ASM ops to create a few-KB copy, and we *might* do an "premature optimization" here.